jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -811,6 +870,16 @@ Object runNativeMethod(Addressable downcallAddr, SegmentAllocator segmtAllocator
 		releaseScope();
 		/*[ENDIF] JAVA_SPEC_VERSION >= 21 */
 
+		/*[IF JAVA_SPEC_VERSION >= 22]*/
+		/* Reset the heap argument information for the current thread so as to
+		 * update the base object of different heap arguments passed by the same
+		 * downcall handler within the current thread.
+		 */
+		if (heapArgInfo != null) {
+			threadHeapArgInfoMap.put(currentThread, new HeapArgInfo(heapBases, heapOffsets, 0));
+		}
```

Comment:
We don't need to allocate a new `HeapArgInfo` object here, we only need to release the existing instance.
```java
		if (info != null) {
			heapArgInfo.remove();
		}
```

Reply:
Updated the code with `ThreadLocal` and verified in tests.